pkg/vsphere/session/session.go

```diff
@@ -193,45 +195,55 @@ func (s *Session) Populate(ctx context.Context) (*Session, error) {
 
 	finder := s.Finder
 
+	log.Debug("vSphere resource cache populating...")
 	s.Datacenter, err = finder.DatacenterOrDefault(ctx, s.DatacenterPath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding dc (%s): %s", s.DatacenterPath, err.Error()))
 	} else {
 		finder.SetDatacenter(s.Datacenter)
+		log.Debugf("Cached dc: %s", s.DatacenterPath)
 	}
 
+	finder.SetDatacenter(s.Datacenter)
+
 	s.Cluster, err = finder.ComputeResourceOrDefault(ctx, s.ClusterPath)
 	if err != nil {
-		errs = append(errs, err.Error())
+		errs = append(errs, fmt.Sprintf("Failure finding cluster (%s): %s", s.ClusterPath, err.Error()))
 	}
+	log.Debugf("Cached cluster: %s", s.ClusterPath)
```

Comment:
These Debug's should be within an else clause, since we didn't cache foo if there was an error.